src/main/scala/mesosphere/marathon/FeaturesConf.scala

```diff
@@ -32,13 +31,13 @@ trait FeaturesConf extends ScallopConf {
     validate = validateFeatures
   )(setStringParser)
 
-  lazy val deprecatedFeatures = opt[Set[DeprecatedFeature]](
+  lazy val deprecatedFeatures = opt[DeprecatedFeatureSet](
     "deprecated_features",
     descr = "A comma-separated list of deprecated features to continue to enable",
     required = false,
-    default = Some(Set.empty),
+    default = Some(DeprecatedFeatureSet(BuildInfo.version, Set.empty)),
     noshort = true,
-    validate = validateDeprecatedFeatures)(deprecatedFeatureParser)
+    validate = { dfs => dfs.isValid() })(deprecatedFeatureParser)
```

Comment:
I tried going through `ScallopConf` source code to find out what exactly happens when valueconverter throws exception (which is what we do with the require) but I guess it will blow up so we should be fine :)

Reply:
Hmm, `.isValid` doesn't throw an exception, it just returns false. A param failing validation will cause Marathon to not launch